### PR TITLE
Update good practices when committing

### PR DESCRIPTION
**Pull request by TessFerrandez**
The readme contained a TODO section about commit messages. This PR replaces this TODO with content about good commits

- Add information about good commits (small, tested etc.)
- Add information about what a commit/PR message should answer
- Move the section around the structure of commit messages from git.md to readme.md
- Add considerations for good commits
- Add references

### CODE REVIEW

File: source-control/readme.md [hunk 1]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
 
 ## Commit Best Practices
 
-TODO: Add more content and details here!
+* Make small commits. This makes changes easier to review, and if we need to revert a commit, we loose less work.
```
Comment (rozele) [inline source-control/readme.md]: `loose` -> `lose`

File: source-control/readme.md [hunk 2]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
...
+* Commit complete and well tested code. Never commit incomplete code, get in the habit of testing your code before commiting.
```
Comment (rozele) [inline source-control/readme.md]: `commiting` -> `committing`

File: source-control/readme.md [hunk 3]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
...
+* Don't mix whitespace changes with functional code changes. It is hard to determine if the line has a functional change or only removes a whitespace, so funcitonal changes may go unnoticed.
```
Comment (rozele) [inline source-control/readme.md]: `funcitonal` -> `functional`

File: source-control/readme.md [hunk 4]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
...
+* Write good commit messages.
 
-* Link work items
-* How often to commit
-* When to push
+A good commit message should answer these questions:
+
+* Why is it neccessary? It may fix a bug, add a feature, improve performance, or just be a change for the sake of correctness
```
Comment (rozele) [inline source-control/readme.md]: `neccessary` -> `necessary`

File: source-control/readme.md [hunk 5]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
...
+* How does it adress the issue? For short, obviuos changes, this can be omitted
```
Comment (rozele) [inline source-control/readme.md]: `adress` -> `address`.

File: source-control/readme.md [hunk 6]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
...
+* What effects does this change have? In addition to the obvious ones, this may include benchmarks, side effects etc.
+* What limitations does the current code have?
+
+Consider this when writing your commit message:
+
+* Don't assume that the code is self-evident/self-documenting
+* If it seems difficult to summarize your commit, it may be because it includes more than one logical change or bug fix. If so, it is better to split it into separate commits with `git add -p`
+* Don't assume the reviewer understands the original problem.
+  It should be possible to review a change request without reading the contents of the original bug/task.
```
Comment (rozele) [inline source-control/readme.md]: Remove newline between `original problem.` and `It should be... `.

File: source-control/readme.md [hunk 7]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
...
+
+Good message structure:
+
+* Separate subject from body with a blank line
+* Limit the subject line to 50 characters
+* Capitalize the subject line
+* Do not end the subject line with a period
+* Use the imperative mood in the subject line (*Fix typo in log* vs. *Fixed typo in log* or *Misc fixes in log code*)
```
Comment (tompaana) [inline source-control/readme.md]: Just curious: Where does this come from? Perhaps, I've being doing it wrong (or it's just matter of taste), but I tend to use the 3rd person form "Fixes typo in log" as that's what the change will do.

To be clear, I'm not requesting a change here.

File: source-control/readme.md [hunk 8]
```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
...
+- Helps teams automate linting and build verification for go projects.
+- Also gives a list of items to verify for go code reviews.
+
+The PR does not add info about VS Code extensions for go, this will
+be added in issue #124
+
+Closes: #123
+```
+
+You can specify the default git editor, which allows you to write your commit messages using your favorite editor. The following command makes vscode your default git editor:
```
Comment (rozele) [inline source-control/readme.md]: `vscode` -> `Visual Studio Code`

File: source-control/readme.md (hunk 7 above)
Comment (TessFerrandez) [reply source-control/readme.md]: From here https://chris.beams.io/posts/git-commit/ - using imperative mood is becoming a bit of a de facto standard I think... This particular path was just moved from the git doc.  I personally am not a big fan of the long messages - but left it there as the Chris Beams suggestions seem to be what most people recommend

File: source-control/readme.md (hunk 1 above)
Comment (TessFerrandez) [reply source-control/readme.md]: Apologies, lots of typos in here, wish I had a typo extension in VS Code :) will fix them all

Comment (TessFerrandez) [reply source-control/readme.md]: fixed

File: source-control/readme.md (hunk 3 above)
Comment (TessFerrandez) [reply source-control/readme.md]: fixed

**Pull request merged**